tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/asr_encoder_prediction_joint_evaluator.py

```diff
@@ -0,0 +1,669 @@
+"""
+Copyright (c) 2018-2021 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from pathlib import Path
+import pickle
+from functools import partial
+from collections import OrderedDict
+import numpy as np
+
+from ..base_evaluator import BaseEvaluator
+from ..quantization_model_evaluator import create_dataset_attributes
+from ...adapters import create_adapter
+from ...config import ConfigError
+from ...launcher import create_launcher
+from ...utils import contains_all, contains_any, extract_image_representations, read_pickle, get_path
+from ...progress_reporters import ProgressReporter
+from ...logging import print_info
+
+
+class AutomaticSpeechRecognitionEvaluator(BaseEvaluator):
```

Comment:
I do not see the difference in evaluator code between this and encoder-decoder (all diff incapsulated in the model), maybe you can just inherit it from that class for avoiding a lot of code duplication?

Reply:
please also provide docs for your evaluator

Reply:
Fixed